src/layouts/EditHomepage/EditHomepage.jsx

```diff
@@ -267,6 +291,32 @@ const EditHomepage = ({ match }) => {
             sectionsErrors.push({ infopic: getErrorValues(INFOPIC_SECTION) })
           }
 
+          if (section.announcements) {
+            sectionsErrors.push({
+              announcements: getErrorValues(ANNOUNCEMENT_BLOCK),
+            })
+            announcementItemErrors = _.map(
+              section.announcements.announcement_items,
+              () => getErrorValues(ANNOUNCEMENT_SECTION)
+            )
+            if (!section.announcements.announcement_items) {
+              // define an empty array to announcement_items to prevent error
+              frontMatter = update(frontMatter, {
+                sections: {
+                  [frontMatter.sections.findIndex((section) =>
+                    EditorHomepageFrontmatterSection.isAnnouncements(section)
+                  )]: {
+                    announcements: {
+                      announcement_items: {
+                        $set: [],
+                      },
+                    },
+                  },
+                },
+              })
+            }
+          }
```

Comment:
seems like we're following the style of the existing code here rgiht?

Reply:
yup, this is correct